src/domain/model/test/interpretation/Condition.test.ts

```diff
@@ -3,14 +3,14 @@ import { Condition } from './Condition';
 describe('Condition', () => {
   describe('SimpleCondition', () => {
     it('does not accept an invalid condition', () => {
-      expect(() => Condition.from({})).toThrow();
-      expect(() => Condition.from({ property: null, value: 3, comparator: '==' })).toThrow();
-      expect(() => Condition.from({ property: 'a', value: null, comparator: '' })).toThrow();
-      expect(() => Condition.from({ property: 'a', value: 3, comparator: '' })).toThrow();
+      expect(() => Condition.fromSchema({})).toThrow();
+      expect(() => Condition.fromSchema({ property: null, value: 3, comparator: '==' })).toThrow();
+      expect(() => Condition.fromSchema({ property: 'a', value: null, comparator: '' })).toThrow();
+      expect(() => Condition.fromSchema({ property: 'a', value: 3, comparator: '' })).toThrow();
```

Comment:
Why 'schema', there's no use of JSON schema here, right?

Reply:
I just had an idea...  Maybe the rules could be JSON schema.  In JSON schema you have 'allOf' which is equivalent to 'and', and 'oneOf' which is equivalent to 'or'.  Both contain an array of child schemas.  If something can only take a specific value that is represented by 'enum: ["myValue"]' or const: "myValue".  Minimum and maximum values are also permitted. E.g. { type: "number", min: 5, max: 10 }.  Can't think of a way to do not equal 🤔

If the schema validates, the rule is satisfied.

Reply:
Also 'and' is implicit when validating a schema, so igg, for example, would be:
```
{
  type: "object",
  properties: {
    c: { const: true },
    igg: { const: true }
  }
}
```

Reply:
Not syntax for not equals https://stackoverflow.com/questions/53338039/json-schema-to-ensure-a-string-does-not-match-any-element-of-an-array

Reply:
This makes a lot of sense 👌 Great idea! I updated the PR to use JSON schema instead